### ISSUE TITLE
0.17 Crash log Scenario

### ISSUE THREAD

**Issue by IIPoliII** [Oarcinae/FactorioScenarioMultiplayerSpawn#64]
```
20466.586 Error MainLoop.cpp:1128: Exception at tick 2956361: The scenario level caused a non-recoverable error.
Please report this error to the scenario author.

Error while running event level::on_entity_spawned (ID 129)
position not specified.
stack traceback:
        ...ctrserver/temp/currently-playing/lib/separate_spawns.lua:219: in function 'ModifyEnemySpawnsNearPlayerStartingAreas'
        /home/fctrserver/temp/currently-playing/control.lua:308: in function </home/fctrserver/temp/currently-playing/control.lua:306>
stack traceback:
        [C]: in function 'create_entity'
        ...ctrserver/temp/currently-playing/lib/separate_spawns.lua:219: in function 'ModifyEnemySpawnsNearPlayerStartingAreas'
        /home/fctrserver/temp/currently-playing/control.lua:308: in function </home/fctrserver/temp/currently-playing/control.lua:306
20466.594 Error ServerMultiplayerManager.cpp:92: MultiplayerManager failed: "The scenario level caused a non-recoverable error.
Please report this error to the scenario author.

Error while running event level::on_entity_spawned (ID 129)
position not specified.
stack traceback:
        ...ctrserver/temp/currently-playing/lib/separate_spawns.lua:219: in function 'ModifyEnemySpawnsNearPlayerStartingAreas'
        /home/fctrserver/temp/currently-playing/control.lua:308: in function </home/fctrserver/temp/currently-playing/control.lua:306>
stack traceback:
        [C]: in function 'create_entity'
        ...ctrserver/temp/currently-playing/lib/separate_spawns.lua:219: in function 'ModifyEnemySpawnsNearPlayerStartingAreas'
        /home/fctrserver/temp/currently-playing/control.lua:308: in function </home/fctrserver/temp/currently-playing/control.lua:306>
20466.594 Info ServerMultiplayerManager.cpp:714: updateTick(2956361) changing state from(InGame) to(Failed)
20466.594 Quitting: multiplayer error.
20466.594 Info GlobalContext.cpp:679: Waiting for child processes to exit:
20467.417 Info ServerMultiplayerManager.cpp:138: Quitting multiplayer connection.
20467.417 Info ServerMultiplayerManager.cpp:714: updateTick(4294967295) changing state from(Failed) to(Closed)
20467.418 Info HttpSharedState.cpp:54: Downloading https://multiplayer.factorio.com/remove-game/4694474
20467.517 Info UDPSocket.cpp:210: Closing socket

```

**Comment by Oarcinae** [comment]
Oh no! I'll look into it today and add some checks to catch this kind of thing. Is it easy to reproduce? Does it happen quite quickly?

**Comment by IIPoliII** [comment]
Awh sorry to hear this :/ sadly not was after 15 hours of game. Want me to send you the full log if I found it ?

**Comment by IIPoliII** [comment]
I watched the log sadly there is not a lot of Infos in it I will ask the player if they saw something

**Comment by Oarcinae** [comment]
I think you provided enough info, I just have to figure out some possible causes and add in a check or two to make sure it doesn't happen.

**Comment by IIPoliII** [comment]
Okay thanks can you comment and close this thread one it's done :3

**Comment by Oarcinae** [comment]
Found the bug. Should be fixed now. I was able to repro and fix. It was a really stupid bug, I thought I had tested that function but clearly I hadn't. Sorry!

Please re-open if you see it again, but I'm pretty sure I got it.

**Comment by IIPoliII** [comment]
Sure thanks for the fix and taking it like this :smile: that was fast too

**Comment by IIPoliII** [comment]
Oh but how did you reproduced it I am interested now :thinking:

**Comment by Oarcinae** [comment]
Well the function in question is called when evolution is higher, and when biters/spitters spawn near a player base. Anything inside warning/danger zones are scaled down a bit so that any player joining won't be attacked by behemoth biters close by if evolution is high.

To reproduce, I just set evolution to max (1.00) and then dumped a large amount of pollution and waited for some biters to spawn near my base. Took me like 5 minutes to figure out my silly mistake. Ended up that I just didn't name the variable correctly in the function :)
